PyStemmusScope/bmi/implementation.py

```diff
@@ -101,16 +109,21 @@ def set_variable(
     else:
         vals = value
```

Comment:
can you please add a check with an error message that the size of `value` should be equal to the size of `vals`.

Reply:
I've added checks for this in the `set_value` and `set_value_at_indices` methods of the BMI.